src/udev/udev-builtin-net_id.c

```diff
@@ -566,6 +566,54 @@ static int pci_get_hotplug_slot(sd_device *dev, uint32_t *ret) {
         return -ENOENT;
 }
 
+static bool get_sun(sd_device *dev, uint32_t *ret) {
```

Comment:
please just propagate the original error and let the caller then treat that as readong to ignore this device or

Reply:
also, please give this function a slightly better name

Reply:
get_device_firmware_node_sun ?